From f1bd917a8f842f523543406023202e2be06f3020 Mon Sep 17 00:00:00 2001
From: Haseeb Ashraf <haseeb_ashraf@mentor.com>
Date: Fri, 14 Mar 2025 13:13:29 +0500
Subject: [PATCH 1/1] cube/CMakeLists.txt: locate wayland-scanner native binary
 properly

This fixes error for wrong wayland-scanner path
```
Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: VERBOSE=1 cmake --build /data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/build --target all --
| Change Dir: '/data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/build'
|
| Run Build Command(s): ninja -v -j 64 all
| ninja: error: '/usr/bin/wayland-scanner', needed by 'cube/xdg-decoration-client-header.h', missing and no known rule to make it
|
| WARNING: /data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/temp/run.do_compile.3853217:153 exit 1 from 'eval ${DESTDIR:+DESTDIR=${DESTDIR} }VERBOSE=1 cmake --build '/data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/build' "$@" -- ${EXTRA_OECMAKE_BUILD}'
| WARNING: Backtrace (BB generated script):
| 	#1: cmake_runcmake_build, /data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/temp/run.do_compile.3853217, line 153
| 	#2: cmake_do_compile, /data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/temp/run.do_compile.3853217, line 147
| 	#3: do_compile, /data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/temp/run.do_compile.3853217, line 142
| 	#4: main, /data/hashraf/innexis_cb_2024.04.0/build-iana/tmp/work/armv8a-oe-linux/vulkan-tools/1.3.296.0/temp/run.do_compile.3853217, line 166
ERROR: Task (/data/hashraf/innexis_cb_2024.04.0/oe-core/meta/recipes-graphics/vulkan/vulkan-tools_1.3.296.0.bb:do_compile) failed with exit code '1'
```

Upstream-Status: Inappropriate [oe-core specific]

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>
---
 cube/CMakeLists.txt | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/cube/CMakeLists.txt b/cube/CMakeLists.txt
index 880cf741..f92f4445 100644
--- a/cube/CMakeLists.txt
+++ b/cube/CMakeLists.txt
@@ -81,7 +81,15 @@ if (CMAKE_SYSTEM_NAME MATCHES "Linux|BSD|GNU")
     if(BUILD_WSI_WAYLAND_SUPPORT)
         pkg_check_modules(WAYLAND_CLIENT REQUIRED IMPORTED_TARGET wayland-client)
 
-        pkg_get_variable(WAYLAND_SCANNER_EXECUTABLE wayland-scanner wayland_scanner)
+        pkg_get_variable(WAYLAND_SCANNER_VAR wayland-scanner wayland_scanner)
+        # Do not assume WAYLAND_SCANNER_VAR to be absolute path and instead take the path as hint
+        # and use find_program() to locate the executable under CMAKE_FIND_ROOT_PATH
+        get_filename_component(WAYLAND_SCANNER_DIR  ${WAYLAND_SCANNER_VAR} DIRECTORY)
+        get_filename_component(WAYLAND_SCANNER_NAME ${WAYLAND_SCANNER_VAR} NAME)
+        find_program(WAYLAND_SCANNER_EXECUTABLE "${WAYLAND_SCANNER_NAME}" PATHS "${WAYLAND_SCANNER_DIR}" NO_DEFAULT_PATH)
+        if(NOT DEFINED WAYLAND_SCANNER_EXECUTABLE)
+            message(FATAL_ERROR "Unable to locate ${WAYLAND_SCANNER_NAME}")
+        endif()
         message(STATUS "WAYLAND_SCANNER_EXECUTABLE = ${WAYLAND_SCANNER_EXECUTABLE}")
 
         pkg_get_variable(WAYLAND_PROTOCOLS_PATH wayland-protocols pkgdatadir)
-- 
2.34.1

